woocommerce/woocommerce-gutenberg-products-block
# Abstract block refactor

Targetting https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/655 with this PR as it follows on.

Creates new Abstract block classes so each block has a PHP class and all follow the same standards. There are 3 abstracts; `AbstractBlock`, `AbstractDynamicBlock`, and `AbstractProductGrid`. 

Functionality is unchanged. This is a refactor.

To test, 

- Checkout branch
- Run `composer update`
- Run `npm run build`
- spot check that each block type is still working. That includes the new Product Categories block.
